From mihnea@zulu.ro Fri Jul 29 04:20:04 2005
Date: Fri, 29 Jul 2005 13:48:48 +0300 (EEST)
From: Mihnea-Costin Grigore <mihnea@zulu.ro>
To: Phil Dibowitz <phil@ipom.com>, Greg Kroah-Hartman <greg@kroah.com>
Subject: usb-storage: Add IGNORE_RESIDUE flag for Mitsumi USB 2.0 card reader (VIA hardware)
Message-ID: <Pine.LNX.4.62.0507291331160.8225@tenebus.tenebus.net>


This patch adds an entry in the unusual_devs.h file for a Mitsumi card
reader/floppy combo that uses a VIA chipset. The IGNORE_RESIDUE flag was
needed for the second LUN to operate properly.

Signed-off-by: Mihnea-Costin Grigore <mihnea@zulu.ro>
Signed-off-by: Phil Dibowitz <phil@ipom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/storage/unusual_devs.h |    7 +++++++
 1 files changed, 7 insertions(+)

--- gregkh-2.6.orig/drivers/usb/storage/unusual_devs.h	2005-08-08 15:53:29.000000000 -0700
+++ gregkh-2.6/drivers/usb/storage/unusual_devs.h	2005-08-08 15:53:32.000000000 -0700
@@ -79,6 +79,13 @@
 		US_SC_8070, US_PR_SCM_ATAPI, init_usbat, 0), 
 #endif
 
+/* Patch submitted by Mihnea-Costin Grigore <mihnea@zulu.ro> */
+UNUSUAL_DEV(  0x040d, 0x6205, 0x0003, 0x0003,
+		"VIA Technologies Inc.",
+		"USB 2.0 Card Reader",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_IGNORE_RESIDUE ),
+
 /* Deduced by Jonathan Woithe <jwoithe@physics.adelaide.edu.au>
  * Entry needed for flags: US_FL_FIX_INQUIRY because initial inquiry message
  * always fails and confuses drive.
